src/components/Settings/Speech/Speech.component.js

```diff
@@ -107,27 +119,35 @@ const Speech = ({
                 max={MAX_RATE}
                 step={INCREMENT_RATE}
                 onChange={handleChangeRate}
+                disabled={voice.voiceSource === 'cloud' ? true : false}
               />
             </div>
           </ListItem>
         </List>
       </Paper>
-      <Menu
-        id="voice-menu"
-        anchorEl={anchorEl}
-        open={voiceOpen}
-        onClose={handleVoiceClose}
-      >
-        {langVoices.map((voice, index) => (
-          <MenuItem
-            key={index}
-            selected={index === selectedVoiceIndex}
-            onClick={() => handleMenuItemClick(voice, index)}
-          >
-            {getVoiceLabel(voice)}
-          </MenuItem>
-        ))}
-      </Menu>
+      {langVoices.length && (
+        <Menu
+          id="voice-menu"
+          anchorEl={anchorEl}
+          open={voiceOpen}
+          onClose={handleVoiceClose}
+        >
+          {langVoices.map((voice, index) => (
+            <MenuItem
+              key={index}
+              selected={index === selectedVoiceIndex}
+              onClick={() => onMenuItemClick(voice, index)}
```

Comment:
we should prevent the selection of an "online" voice if the redux state `app.isConnected` is false.
I mean if we don't prevent this. Voice would be undefined and the communicator hasn't voice
I think that is a good option show a notification and prevent the fire `onMenuItemClick(voice, index)`.
I could be wrong.